### PR TITLE
Improve error handling + add auto refresh + other small changes

### PR DESCRIPTION
**Pull request by raymondtruong**
This PR contains the following changes:
- Error handling:
  - added a banner when an error occurs when generating SKU recommendations:
  image
  - added error handling for the case where backend SKU recommendation call returns a result without any errors, but the result is incomplete because no SKUs could be recommended (e.g. source is too big for any possible SKUs)
  - added a backend call which will fetch the latest errors that occurred during data collection and display them as ADS notifications, if any (see below)

- Auto-refresh during perf data collection:
  - 1) after starting perf data collection, periodically new SKU recommendations will be generated (this should be every 10 minutes or so)
  - 2) after starting perf data collection, periodically the new backend call mentioned above will fetch for any errors during the collection progress (this should be every 15/30 seconds or so). This will also refresh the text shown indicating the time elapsed/left on the timer from point a

- Other changes:
  - update SKU recommendation contract to pass "include preview SKUs?" flag set by UI
  - update source properties table to point to the correct fields, format all values to two decimal points, and show 'N/A' for IO latency row when IO throughput is below the idle threshold
  - update strings

### CODE REVIEW

**Review by neetusingh-ms — APPROVED** [review]

File: extensions/sql-migration/src/dialog/skuRecommendationResults/skuRecommendationResultsDialog.ts [hunk 1]
```diff
@@ -432,12 +432,12 @@ export class SkuRecommendationResultsDialog {
...
-		const logStorageRow = createRow(constants.LOG_STORAGE_REQUIREMENT, constants.STORAGE_GB(instanceRequirements?.logStorageRequirementInMB! / 1024));
+		const memoryRow = createRow(constants.MEMORY_REQUIREMENT, constants.GB(instanceRequirements?.memoryRequirementInMB! / 1024));
+		const dataStorageRow = createRow(constants.DATA_STORAGE_REQUIREMENT, constants.GB(instanceRequirements?.dataStorageRequirementInMB! / 1024));
+		const logStorageRow = createRow(constants.LOG_STORAGE_REQUIREMENT, constants.GB(instanceRequirements?.logStorageRequirementInMB! / 1024));
 		const dataIOPSRow = createRow(constants.DATA_IOPS_REQUIREMENT, constants.IOPS(instanceRequirements?.dataIOPSRequirement!));
 		const logsIOPSRow = createRow(constants.LOGS_IOPS_REQUIREMENT, constants.IOPS(instanceRequirements?.logIOPSRequirement!));
-		const ioLatencyRow = createRow(constants.IO_LATENCY_REQUIREMENT, constants.MS(instanceRequirements?.ioLatencyRequirementInMs!));
+		const ioLatencyRow = createRow(constants.IO_LATENCY_REQUIREMENT, instanceRequirements?.ioThroughputRequirementInMBps! < 5 ? 'N/A' : constants.MS(instanceRequirements?.ioLatencyRequirementInMs!));
```
Comment (rchlkm) [inline extensions/sql-migration/src/dialog/skuRecommendationResults/skuRecommendationResultsDialog.ts:440]: can we make the N/A a localized string?

File: extensions/sql-migration/src/models/stateMachine.ts [hunk 2]
```diff
@@ -393,6 +407,48 @@ export class MigrationStateModel implements Model, vscode.Disposable {
 
 			this._startPerfDataCollectionApiResponse = response!;
 			this._perfDataCollectionStartDate = this._startPerfDataCollectionApiResponse.dateTimeStarted;
+
+			const classVariable = this;
+			console.log('starting auto refresh for perf collection status update with interval ' + this.refreshPerfDataCollectionFrequency);
```
Comment (rchlkm) [inline extensions/sql-migration/src/models/stateMachine.ts:412]: let's make sure to clean up all the console logs

File: extensions/sql-migration/src/models/stateMachine.ts [hunk 3]
```diff
@@ -393,6 +407,48 @@ export class MigrationStateModel implements Model, vscode.Disposable {
...
+			clearInterval(this._autoRefreshPerfDataCollectionHandle);
+			if (this.refreshPerfDataCollectionFrequency !== -1) {
+				this._autoRefreshPerfDataCollectionHandle = setInterval(async function () {
+					await classVariable.refreshPerfDataCollection();
+					await page.refreshDataCollectionTimerStatus();				// update timer
+
+					// TO-DO: state machine will create an ADS notification (bottom right corner)
```
Comment (rchlkm) [inline extensions/sql-migration/src/models/stateMachine.ts:419]: I can create the ADS popup notification. I don't think a banner is necessary

**Review by rchlkm — CHANGES_REQUESTED** [review]

File: extensions/sql-migration/src/models/stateMachine.ts (hunk 3 above)
Comment (raymondtruong) [reply extensions/sql-migration/src/models/stateMachine.ts:419]: Oops, I already did that and forgot to take the comment out

**Review by rchlkm — APPROVED** [review]

**Pull request merged**